File: backend/src/main/kotlin/com/example/backend/models/ProduitCmd.kt
package com.example.backend.models

import jakarta.persistence.*
import org.hibernate.annotations.ColumnDefault

@Entity
@Table(name = "produit_cmd")
class ProduitCmd {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "id", nullable = false)
  var id: Int? = null

  @Column(name = "prixPublic")
  var prixPublic: Double? = 0.0

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "produit_id", insertable = false, updatable = false)
  var produit: Produit? = null

  @Column(name = "produit_id")
  var produitId: Int? = null

  @Column(name = "commande_id")
  var commandeId: Long? = null

  @ManyToOne(fetch = FetchType.EAGER)
  @JoinColumn(name = "commande_id", insertable = false, updatable = false)
  var commande: Commande?=null

  @Column(name = "puCmd")
  var puCmd: Double? = 0.0

  @Column(name = "ptCmd")
  var ptCmd: Double? = 0.0

  @Column(name = "qtiteCmd")
  var qtiteCmd: Int? = null

  @Column(name = "puRecept")
  var puRecept: Double? = 0.0

  @Column(name = "ptRecept")
  var ptRecept: Double? = 0.0

  @Column(name = "qtiteRecu")
  var qtiteRecu: Int? = null

  @Column(name = "uniteGratuite")
  var uniteGratuite: Int? = null

  @Column(name = "etat")
  var etat: String? = null

  @ColumnDefault("0")
  @Column(name = "supprimer")
  var  supprimer: Int? = 0
}

File: backend_py/pharmaPython/app/models/produit_cmd.py
from sqlalchemy import Column, Integer, Float, String, ForeignKey
from sqlalchemy.orm import relationship
from app.core.db import Base


class ProduitCmd(Base):
  __tablename__ = "produit_cmd"
  __table_args__ = {"extend_existing": True}

  id = Column(Integer, primary_key=True, autoincrement=True)

  prix_public = Column(Float, default=0.0)

  # >>> AJOUTE les ForeignKey explicites <<<
  produit_id = Column(Integer, ForeignKey("produit.id"), nullable=False)
  commande_id = Column(Integer, ForeignKey("commande.id"), nullable=False)

  pu_cmd = Column(Float, default=0.0)
  pt_cmd = Column(Float, default=0.0)
  qtite_cmd = Column(Integer)

  pu_recept = Column(Float, default=0.0)
  pt_recept = Column(Float, default=0.0)
  qtite_recu = Column(Integer)
  unite_gratuite = Column(Integer)

  etat = Column(String)
  supprimer = Column(Integer, default=0)

  # >>> ALIGNE les relations avec les autres modèles <<<
  # Si Commande définit: produits_cmd = relationship("ProduitCmd", back_populates="commande")
  commande = relationship("Commande", back_populates="produits_cmd")

  produit = relationship("Produit")
  # Si Produit définit: produits_cmd = relationship("ProduitCmd", back_populates="produit")
  # produit = relationship("Produit", back_populates="produits_cmd")

  # produit = relationship("Produit", foreign_keys=[produit_id])
  # produit = relationship("Produit", foreign_keys=[produit_id])
  # commande = relationship("Commande", foreign_keys=[commande_id])
  # commande = relationship("Commande", foreign_keys=[commande_id])
  #
  # produit = relationship("Produit", back_populates="produits_cmd")
